chrome/content/zotero/elements/abstractBox.js

```diff
@@ -107,10 +99,9 @@
 			}
 		}
 
-		render() {
-			if (!this.item) {
-				return;
-			}
+		render(force = false) {
```

Comment:
When would `force` be necessary? If we need it, we should document somewhere what it does and why it exists.

Reply:
The idea here is, basically the `render` is only called by the CE's container, which decides when it should be rendered, e.g. when scrolling into view. However, the container only knows when it should be reminded to render, but doesn't know if it's already rendered. So by default, the `force` is always `false`, and duplicate rendering calls are ignored, which is secured by `_isAlreadyRendered`. Say if you keep scrolling a section into and out of view, the `render` will be triggered every time but only the first one renders the section.

However, when it comes to notifier callbacks and other internal calls from this CE itself, that's where we want a force render to happen, no matter if it's already rendered.

I will update the implementation to make it more clear.

Reply:
Anyway, seems like we don't have a better choice here. So let's pass them into itemPane.